fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -304,10 +304,11 @@ public boolean skipUpdateForUnchangedKeys() {
      * @param savedRecord the record being indexed
      * @param remove <code>true</code> if removing from index
      * @param indexEntry the entry for the index to be updated
+     * @return a future completed when the key is updated
      */
-    protected <M extends Message> void updateOneKey(@Nonnull final FDBIndexableRecord<M> savedRecord,
-                                                    final boolean remove,
-                                                    @Nonnull final IndexEntry indexEntry) {
+    protected <M extends Message> CompletableFuture<Void> updateOneKeyAsync(@Nonnull final FDBIndexableRecord<M> savedRecord,
```

Comment:
I think I'm ok with this change, but I do know of at least one down-stream project that overrides `updateOneKey`.  If we were really concerned about compatibility, this could be changed to something like:

```
protected <M extends Message> CompletableFuture<Void> updateOneKeyAsync(@Nonnull final FDBIndexableRecord<M> savedRecord,
                                                                        final boolean remove,
                                                                        @Nonnull final IndexEntry indexEntry) {
    updateOneKey(savedRecord, remove, indexEntry);
    return postUpdateOneKeyAsync(savedRecord, remove, indexEntry).
}
```

This is kind of convoluted and gross though, so I'm not strongly advocating for it....just that it is a consideration.